src/openvic-simulation/dataloader/Dataloader.cpp

```diff
@@ -824,6 +828,46 @@ bool Dataloader::_load_map_dir(DefinitionManager& definition_manager) const {
 	return ret;
 }
 
+bool Dataloader::_load_song_chances(DefinitionManager& definition_manager, fs::path song_chance_file) {
+	bool ret = true;
+	SongChanceManager& song_chance_manager = definition_manager.get_song_chance_manager();
+
+	if(song_chance_file.empty()){
+		Logger::info("No Songs.txt file to load");
+		song_chance_manager.lock_songChances();
+		return true;
+	}
+
+	ret &= song_chance_manager.load_songs_file(parse_defines_cached(song_chance_file).get_file_node());
+
+	song_chance_manager.lock_songChances();
+
+	return ret;
+}
+
+bool Dataloader::_load_sfx_defines(DefinitionManager& definition_manager, fs::path sfx_file){
+	bool ret = true;
+	SfxManager& sfx_manager = definition_manager.get_sfx_manager();
+
+	if(sfx_file.empty()){
+		Logger::info("No sound.sfx file to load");
+		sfx_manager.lock_sfx_defines();
+		return false;
+	}
+
+	if (!sfx_manager.load_sound_defines_file(
+		parse_defines_cached(sfx_file).get_file_node()
+	)) {
+		Logger::error("Error while loading sound defines file!");
+		ret = false;
+	}
+
+	sfx_manager.lock_sfx_defines();
+
+	return ret;
+
+}
```

Comment:
```suggestion
bool Dataloader::_load_sfx_defines(DefinitionManager& definition_manager) const {
	static constexpr std::string_view sfx_file = "interface/sound.sfx";

	bool ret = true;
	SfxManager& sfx_manager = definition_manager.get_sfx_manager();

	ret &= sfx_manager.load_sound_defines_file(parse_defines(lookup_file(sfx_file)).get_file_node());

	sfx_manager.lock_sfx_defines();

	return ret;
}
```
This function can be const as we don't need to cache the parser (it can safely be freed once the load function ends). We can also provide the filepath as a constant, rather than an argument, as it'll always be the same.
There's no need for a check on the looked up filepath being empty, as `lookup_file` will print an error if it isn't found and the expect functions in `load_sound_defines_file` will handle the resulting null filenode properly.
We also don't need to print an error message here if `load_sound_defines_file` fails, as we print an identical message straight afterwards in `load_defines` if this function returns false.